docs: adds async checks docs (#1193)
apollographql/apollo-cli: docs/source/commands/subgraphs.mdx

To configure the behavior of schema checks (such as the time range of past operations to check against), see the [documentation for schema checks](/studio/check-configurations/#using-apollo-studio-recommended).

> If you don't want to wait for the check to complete, you can run the command with the `--background` flag and the check will appear in Apollo Studio on the checks tab.

Reviewer: ```suggestion
If you don't want to wait for the check to complete, you can run the command with the `--background` flag. You can then look up the check's result in Apollo Studio on the Checks tab.
```

#### Running checks in CI

> If you want to run checks in CI, you may want to pass the `--background` flag to `rover subgraph check`. This flag will make Rover start the check in the background. If you have [connected Apollo Studio to your GitHub repository](/studio/github-integration/), it will detect the check and add its own status to your PR.

Reviewer: ```suggestion
If you're running schema checks in CI, you might want to pass the `--background` flag to `rover subgraph check`. This flag instructs Rover to initiate schema checks but _not_ await their result. If you've [connected Apollo Studio to your GitHub repository](/studio/github-integration/), the integration detects the checks execution and adds a status to the associated pull request.
```
